### PR TITLE
Hw-3_Databases

### PR DESCRIPTION
**Pull request by Nyskelo**

### CODE REVIEW

File: server.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
 import http from 'http';
+import mongoose from 'mongoose';
+
 import app from './src/app';
-const port = process.env.PORT || 3000;
+
+const PORT = process.env.PORT || 3000;
+const CONNECTION_STRING = 'mongodb+srv://allaoleksyn:0x7EU2u6HoiYxPCS@cluster0.nqxzydv.mongodb.net/NODE_JS';
```
Comment (vasylkivskyy) [inline server.ts]: Do not add sensitive data to GitHub, you can use .env file for example

File: src/conrollers/movies.controllers.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      next(error);
+    }
+  }
+
+  async DeleteMovieById(req: Request, res: Response, next: NextFunction): Promise<Response<Movie> | undefined> {
+    try {
+      const { id } = req.params;
+      const movie = await movieService.DeleteMovieById(id);
+
```
Comment (vasylkivskyy) [inline src/conrollers/movies.controllers.ts:76]: When you delete movie twice or more, you receive 200 OK Status on the second and next attempts

File: src/conrollers/genres.controllers.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      next(error);
+    }
+  }
+
+  async DeleteGenreById(req: Request, res: Response, next: NextFunction): Promise<Response<Genre> | undefined> {
+    try {
+      const { id } = req.params;
+      const genre = await genreService.DeleteGenreById(id);
+
+      res.status(200).json({ respons: { status: 200, message: 'The Genre has been deleted', genre } });
```
Comment (vasylkivskyy) [inline src/conrollers/genres.controllers.ts]: "respons" => "response"
Small correction)

File: src/schemas/genres.schemas.ts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        500: {
+          $ref: '#/components/responses/500',
+        },
+        503: {
+          $ref: '#/components/responses/503',
+        },
+      },
+    },
+    put: {
+      summary: 'Update a movie by id',
```
Comment (vasylkivskyy) [inline src/schemas/genres.schemas.ts]: I think, it should be "update genre by id"

File: src/conrollers/movies.controllers.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  async GetAllMovies(_req: Request, res: Response, next: NextFunction): Promise<Response<Movie[]> | undefined> {
+    try {
+      const movies = await movieService.GetAllMovies();
+      if (!movies.length) {
+        const response = { status: 204, message: 'No movies found, feel free to add a movie :)' };
+        return res.status(204).send({ response });
+      }
+
+      res.json(movies);
```
Comment (vasylkivskyy) [inline src/conrollers/movies.controllers.ts:31]: In response there are fields "__v" in films (and genres too)

Comment (Nyskelo) [reply src/conrollers/movies.controllers.ts:31]: Set the versionKey to false.

File: src/schemas/genres.schemas.ts (hunk 4 above)
Comment (Nyskelo) [reply src/schemas/genres.schemas.ts]: Changed to "update genre by id".

File: src/conrollers/genres.controllers.ts (hunk 3 above)
Comment (Nyskelo) [reply src/conrollers/genres.controllers.ts]: Changed to 'response'

File: src/conrollers/movies.controllers.ts (hunk 2 above)
Comment (Nyskelo) [reply src/conrollers/movies.controllers.ts:76]: Added additional check for movie availability.

File: server.ts (hunk 1 above)
Comment (Nyskelo) [reply server.ts]: Sensitive data has been deleted from GitHub.

**Review by vasylkivskyy — APPROVED** [review]

**Pull request merged**